Source/drlg_l2.cpp

```diff
@@ -903,8 +903,9 @@ void DRLG_L2PlaceRndSet(BYTE *miniset, int rndper)
 			if (found == TRUE) {
 				for (yy = sy - sh; yy < sy + 2 * sh && found == TRUE; yy++) {
 					for (xx = sx - sw; xx < sx + 2 * sw; xx++) {
-						// BUGFIX: yy and xx can go out of bounds
-						if (dungeon[xx][yy] == miniset[kk]) {
+						// BUGFIX: yy and xx can go out of bounds (fixed)
+						if(0 <= xx && xx < DMAXX && 0 <= yy && yy < DMAXY &&

```

Comment:
Seems like a better place to fix this is adding `std::min/max` to the loop initializers and conditions